src/main/java/mate/dao/DriverDaoImpl.java

```diff
@@ -105,8 +110,27 @@ private Driver getDriver(ResultSet resultSet) throws SQLException {
         Long id = resultSet.getObject("id", Long.class);
         String name = resultSet.getString("name");
         String licenseNumber = resultSet.getString("license_number");
-        Driver driver = new Driver(name, licenseNumber);
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
+        Driver driver = new Driver(name, licenseNumber, login, password);
         driver.setId(id);
         return driver;
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT * FROM drivers WHERE login = ? AND is_deleted = FALSE";
+        Driver driver = null;
```

Comment:
we need to create an object(driver) as close as possible to its first use

Reply:
![image](https://user-images.githubusercontent.com/64101459/137704961-f0aa9806-4d2c-4d79-b760-92ee691441f8.png)